#!/usr/bin/env python
PACKAGE = "hydrus"

from dynamic_reconfigure.parameter_generator_catkin import *

RECONFIGURE_TORSION_MODE_FLAG = 0
TORSION_CONSTANT = 1
TORSION_Q_MU = 2
TORSION_Q_MU_D = 3

gen = ParameterGenerator()

gen.add("torsion_mode_flag", bool_t, RECONFIGURE_TORSION_MODE_FLAG, "Torsion Mode Parameters Flag", False)

gen.add("torsion_constant", double_t, TORSION_CONSTANT, "Torsion constant of EOM", 1, 0, 10000)
gen.add("q_mu", double_t, TORSION_Q_MU, "Torsion LQR Q_MU", 1, 0, 10000)
gen.add("q_mu_d", double_t, TORSION_Q_MU_D, "Torsion LQR Q_MU_D", 1, 0, 10000)

exit(gen.generate(PACKAGE, "hydrus", "torsion_mode"))
